src/modules/src/moduleManager.cpp

```diff
@@ -33,9 +31,11 @@ std::shared_ptr<ModuleWrapper> ModuleManager::GetModule(const std::string & name
 }
 
 void ModuleManager::Start() {
-    for (const auto &[_, module] : m_modules) {
-        m_createTask([module]() { module->Start(); });
+    for (const auto &[_, module] : m_modules)
+    {
+        m_taskManager.EnqueueTask([module]() { module->Start(); }, module->Name());
     }
+    m_taskManager.Start(m_modules.size() * 2);
```

Comment:
```suggestion
    m_taskManager.Start(m_modules.size());
```
We only need one thread per module.